packages/loader/container-loader/src/container.ts

```diff
@@ -1762,11 +1763,19 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
 
     private processSignal(message: ISignalMessage) {
         // No clientId indicates a system signal message.
-        if (message.clientId === null) {
+        console.log("here I am in processSignal -- how can I find a way to differentiate between protocol signals and non-protocol ones?");
+        console.log(message);
+        if (message.clientId === null && message.content.type === SignalType.ExternalDataChange) {
+            console.log('I am correctly processing this signal');
+            const local = this.clientId === message.clientId;
+            this.context.processSignal(message, local); // what is local??
```

Comment:
`local` here means that the signal was sent by the same client that is processing it. Exactly one client will run through this with `local = true`, all others would see `local = false`. Something similar happens with ops.

Reply:
Oh, it's right there in the declaration right above. Thanks!